src/ogd/games/THERMOVR/features/TaskCompleteCount.py

```diff
@@ -7,27 +7,31 @@
 
 class TaskCompleteCount(SessionFeature):
 
-    def __init__(self, params:ExtractorParameters, player_id:str):
-        self._player_id = player_id
+    def __init__(self, params: ExtractorParameters, player_id: str):
+        self.completed_tasks: Set[str] = set()
         super().__init__(params=params)
-        self._task_complete_count = 0
 
     @classmethod
-    def _getEventDependencies(cls, mode:ExtractionMode) -> List[str]:
-        return ["game_state"]
+    def _getEventDependencies(cls, mode: ExtractionMode) -> List[str]:
+        return ["target_state_achieved", "click_submit_answer"]
 
     @classmethod
-    def _getFeatureDependencies(cls, mode:ExtractionMode) -> List[str]:
+    def _getFeatureDependencies(cls, mode: ExtractionMode) -> List[str]:
         return []
 
     def _extractFromEvent(self, event: Event) -> None:
-        if event.UserID == self._player_id:
-            current_task = event.GameState.get('current_task', None)
-            if current_task and current_task.get('is_complete', False):
-                self._task_complete_count += 1
+        if event.EventType == "target_state_achieved":
```

Comment:
Replace `event.EventType` with `event.EventName`

Reply:
Pushed, up for review